src/cli/rustup_mode.rs

```diff
@@ -1578,6 +1586,20 @@ fn set_profile(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
     Ok(utils::ExitCode(0))
 }
 
+fn set_auto_self_update(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
+    if self_update::NEVER_SELF_UPDATE {
+        let mut args = crate::process().args_os();
+        let arg0 = args.next().map(PathBuf::from);
+        let arg0 = arg0
+            .as_ref()
+            .and_then(|a| a.to_str())
+            .ok_or(CLIError::NoExeFilePath)?;
```

Comment:
We already have NoExeName : I don't think is different enough to bother with a new error (and as the error is never caught and never repeated, we can just use anyhow!(...)